pytest_timeout.py

```diff
@@ -193,8 +196,14 @@ def timeout_setup(item):
     params = get_params(item)
     if params.timeout is None or params.timeout <= 0:
         return
-    if params.method == "signal":
 
+    timeout_method = params.method
+    if timeout_method == "signal" and (
+        threading.current_thread() is not threading.main_thread()
+    ):
```

Comment:
```suggestion
    if (timeout_method == "signal"
            and threading.current_thread() is not threading.main_thread()):
```
See https://www.python.org/dev/peps/pep-0008/#should-a-line-break-before-or-after-a-binary-operator